Source/WebKit/NetworkProcess/cache/NetworkCacheIOChannelCocoa.mm

```diff
@@ -96,29 +97,34 @@ static long dispatchQueueIdentifier(WorkQueue::QOS qos)
     RELEASE_ASSERT(!m_wasDeleted.exchange(true));
 }
 
-void IOChannel::read(size_t offset, size_t size, WTF::WorkQueueBase& queue, Function<void(Data&, int error)>&& completionHandler)
+void IOChannel::read(size_t offset, size_t size, Ref<WTF::WorkQueueBase>&& queue, Function<void(Data&&, int error)>&& completionHandler)
 {
-    RefPtr<IOChannel> channel(this);
+    Locker locker { m_lock };
+
     bool didCallCompletionHandler = false;
-    dispatch_io_read(m_dispatchIO.get(), offset, size, queue.dispatchQueue(), makeBlockPtr([channel, completionHandler = WTFMove(completionHandler), didCallCompletionHandler](bool done, dispatch_data_t fileData, int error) mutable {
+    dispatch_io_read(m_dispatchIO.get(), offset, size, queue->dispatchQueue(), makeBlockPtr([protectedThis = Ref { *this }, queue = WTFMove(queue), completionHandler = WTFMove(completionHandler), didCallCompletionHandler](bool done, dispatch_data_t fileData, int error) mutable {
```

Comment:
in C++, the order of evaluation of function parameters is unspecified. This means that the compiler is free to evaluate the arguments in any order it chooses. Therefore, I think the new code is likely unsafe with regards to `queue` which is used and moved here.